### DISCUSSION TITLE
How to add a `namespace` to a class if non is set?

### DISCUSSION THREAD

**Discussion by numediaweb** [rectorphp/rector#8290] [Q&A] [answered]
# Question
I have been playing with this nice library: good job  by the way 👏

Trying to create my own custom rule that adds a `namespace` to a class if non is set.. But I couldn't figure it out sofar...

## The file I have:
```php
<?php

class MyClass extends SecureControllerAction {
...
}
```

## The results I expect:
```php
<?php

namespace App\Controller;

class MyClass extends SecureControllerAction {
...
}
```

## My current rule
```php

    /**
     * @return string[]
     */
    public function getNodeTypes(): array
    {
        return [
            Name::class,
            Class_::class,
            Namespace_::class,
        ];
    }

    /**
     * @param Class_ $node
     */
    public function refactor(Node $node): ?Node
    {
         // Here I'am stuck as the file itself doesn't have a namespace node!
    }
```

Any help would be appreciated 🙏

**Comment by samsonasik** [comment]
You can use `RenameClassRector` for that, see https://getrector.com/demo/e0157fa7-f983-47c7-b443-f54036a16fef

**Reply by numediaweb** [reply]
Thank you so much @samsonasik !
You saved my day ❤️

**Comment by samsonasik** [comment, marked as answer]
@numediaweb FYI, this feature is removed, Rector is now only rename class reference, not class name, see:

- https://github.com/rectorphp/rector-src/pull/5487
- https://github.com/rectorphp/rector-src/pull/5488

for the reasoning
